### DISCUSSION TITLE
requests patch doesn't work even if there are change

### DISCUSSION THREAD

**Discussion by seungtaek** [netbox-community/netbox#11095] [Q&A]
Hello guys, 

I have a question for changing the description of interface.
My logic of code is simple,

1. get ip switch from netbox
2. connect to it using ssh(netmiko)
3. get a name of interface and description and put it to netbox
4. if there are some change in switch(type of inteface or description etc), do update. 

my code works fine from 1 to 3 steps but 4 step doesn't work I couldn't find out because there are even no error message.

> resonse = requests.request("GET", requests_url, headers=HEADERS, data = payload, verify=False)
json_data = resonse.json()
results = json_data['results']
for device in results:
    if(device['device_role']['name'] == 'router-test' and device['site']['name']=='LA' and device['primary_ip'] != None) :
        ipaddr = ipaddress.ip_interface(device['primary_ip']['address']).ip
        hostname = device['name']
        id = device['id']
        print(ipaddr)
        print(hostname)

        backbone_matos = {
            'device_type': 'cisco_ios',
            'host': str(ipaddr),
            'username': 'xxx',
            'password': 'xxxxx'
        }

        commands = [
            "do show interface status",
            ]

        connectssh = ConnectHandler(**backbone_matos)
        connectssh.enable()
        sh_des_all_output = connectssh.send_command('show interfaces status', use_textfsm=True)

        # print(sh_des_all_output)
        print(json.dumps(sh_des_all_output, indent=2))
        for descriptions in sh_des_all_output:
            description = descriptions['name']
            port = descriptions['port']
            print(description)
            print(port)

            Put_url = f"{NB_URL}/api/dcim/interfaces/?device={hostname}"

            interfaces_parameters = {
                "device": id,
                "name": port,
                "description": description,
                "type": "virtual",

            }

            requests.post(Put_url, headers=HEADERS, json=interfaces_parameters, verify=False)
            requests.patch(Put_url, headers=HEADERS, json=interfaces_parameters, verify=False)

> 

Longstory short, 

Last code(requests patch), why it could't update the description??

**Comment by candlerb** [comment]
Firstly, you're not checking the response status. Try:

```
res = requests.patch(...)
res.raise_for_status()
```

If you print the response body (res.text) first, you may also get some more clues.

Secondly: you cannot update a single interface by a patch to `/api/dcim/interfaces/?device={hostname}`. You need to to patch `/api/dcim/interfaces/NNN/` where NNN is the *interface* ID - i.e. the interface detail URL.  This is documented [here](https://docs.netbox.dev/en/stable/integrations/rest-api/#updating-an-object) - see also the inbuilt Swagger API documentation, which you can reach from the "REST API Documentation" button at the bottom of Netbox's own home page.
